client/lib/posts/post-list-cache-store.js

```diff
@@ -1,73 +1,63 @@
 /**
  * External dependencies
  */
-var intersection = require( 'lodash/intersection' );
+import intersection from 'lodash/intersection';
+import debugFactory from 'debug';
 
 /**
  * Internal dependencies
  */
-var sites = require( 'lib/sites-list' )(),
-	Dispatcher = require( 'dispatcher' );
+import sitesFactory from 'lib/sites-list';
+import Dispatcher from 'dispatcher';
 
-var _cache = {},
-	TTL_IN_MS = 5 * 60 * 1000; // five minutes
+const _cache = {};
+const _canonicalCache = {};
+const TTL_IN_MS = 5 * 60 * 1000; // five minutes
+const sites = sitesFactory();
+const PostsListCache = { get: get };
+const debug = debugFactory( 'calypso:posts-list:cache' );
 
 function isStale( list ) {
-	var now = new Date().getTime(),
-		timeSaved = list.timeSaved;
+	const now = new Date().getTime();
+	const { timeSaved } = list;
 	return ( now - timeSaved ) > TTL_IN_MS;
 }
 
-function getCacheKey( options ) {
-	var cacheKey = '',
-		keys = Object.keys( options ).sort();
-
-	keys.forEach( function( key ) {
-		if ( cacheKey.length ) {
-			cacheKey += ':';
-		}
-
-		cacheKey += key + '-' + options[ key ];
-	} );
-
-	return cacheKey;
-}
-
-function get( query ) {
-	var key = getCacheKey( query );
-
-	if ( _cache[ key ] && ! isStale( _cache[ key ] ) && ! _cache[ key ].dirty ) {
-		return _cache[ key ].list;
+function get( listKey ) {
+	if ( _cache[ listKey ] && ! isStale( _cache[ listKey ] ) && ! _cache[ listKey ].dirty ) {
+		return _cache[ listKey ].list;
 	}
 
-	// We assume that when a consumer attempts to access a dirty key that we can safely delete it
-	// because the consumer will get new data to freshen the cache
-	if ( _cache[ key ] && _cache[ key ].dirty ) {
-		delete _cache[ key ];
+	// Delete the dirty cache to force a request for new data
+	if ( _cache[ listKey ] && _cache[ listKey ].dirty ) {
+		debug( 'delete cached list %o', listKey );
+		delete _cache[ listKey ];
```

Comment:
You can modify a constant. It's constant in the sense that the variable is a constant reference to the same object. It doesn't mean that the object itself cannot be changed, but just that the assignment cannot be made again. So this usage is valid. It would be invalid for example if I did something like this:
```js
const example = someObject;
example = omit( example, 'prop' );
```